### ISSUE TITLE
Implement Breadth-First Search (BFS) in C

### ISSUE THREAD

**Issue by sobhanbera** [sobhanbera/algorithms#193]
### Issue Details
**Algorithm/Data Structure**: Breadth-First Search (BFS)
**Programming Language**: C
**Difficulty**: ⭐️⭐️★★★

#### Requirements
- Implement the BFS algorithm in C.
- Ensure that your code is well-documented and easy to understand, especially for beginners.
- Test your implementation with sample graphs to verify its correctness.

#### Additional Information
If you are new to BFS or need guidance on implementing it in C, feel free to ask questions or seek help in the comments. This is a great opportunity for beginners to contribute to an open-source project and learn about graph algorithms in C.

**Note**: Before starting to work on this issue, please comment below to express your interest and let us know if you have any questions or need clarification on the task.

**Comment by MoBanerjee** [comment]
Can you kindly assign me this open issue?
Thanks

**Comment by sobhanbera** [comment]
Yeah @MoBanerjee, I've assigned the issue to you. If need any help do ask here. You can start working on this issue now.
